docs/examples/rx/10_upgrade_chain/index.js

```diff
@@ -0,0 +1,65 @@
+// Import the API & Provider and some utility functions
+const { ApiRx } = require('@polkadot/api');
+const { WsProvider } = require('@polkadot/rpc-provider');
+// import the test keyring (already has dev keys for Alice, Bob, Charlie, Eve & Ferdie)
+const testKeyring = require('@polkadot/keyring/testing');
+const fs = require('fs');
+const { switchMap, first } = require('rxjs/operators');
+
+async function main () {
+  // Initialise the provider to connect to the local node
+  const provider = new WsProvider('ws://127.0.0.1:9944');
+
+  // Create the API and wait until ready (optional provider passed through)
+  const api = await ApiRx.create(provider).toPromise();
+
+  // retrieve the upgrade key from the chain state
+  const adminId = await api.query.sudo.key().toPromise();
+
+  // find the actual keypair in the keyring (if this is an changed value, the key
+  // needs to be added to the keyring before - this assumes we have defaults, i.e.
+  // Alice as the key - and this already exists on the test keyring)
+  const keyring = testKeyring.default();
+  const adminPair = keyring.getPair(adminId.toString());
+
+  // retrieve the runtime to upgrade to
+  const code = fs.readFileSync('./test.wasm').toString('hex');
+  const proposal = api.tx.consensus.setCode(`0x${code}`);
+
+  console.log(`Upgrading chain runtime from ${adminId}`);
+
+  // get the nonce for the admin key
+  api.query.system.accountNonce(adminId).pipe(first(),
+    // pipe nonce into transfer
+    switchMap(adminNonce => api.tx.sudo
+      // preform the actual chain upgrade via the sudo module
+      .sudo(proposal)
+      // sign the proposal
+      .sign(adminPair, adminNonce)
+      // send the proposal
+      .send()))
+    // subscribe to overall result
+    .subscribe(({ events = [], status, type }) => {
+      // Log transfer events
+      console.log('Proposal status:', type);
+
+      if (type === 'Finalised') {
+        console.error('You have just upgraded your chain');
+
+        console.log('Completed at block hash', status.value.toHex());
```

Comment:
`asFinalised` (just noting)

Reply:
I did a replace all now, I didn't realize how often it was actually used.

Reply:
Me neither... new examples are copied of old so it seems to stack up...